crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -314,6 +314,51 @@ impl Schedule {
             system.apply_buffers(world);
         }
     }
+
+    /// Get the state of system-stepping
+    pub fn stepping(&self) -> bool {
+        self.executor.stepping()
+    }
+
+    /// Get the name of the next system that will be run in the
+    /// schedule if stepping is enabled
+    pub fn next_system(&self) -> Option<String> {
+        self.executor
+            .next_system()
+            .map(|idx| self.executable.systems[idx].name().to_string())
```

Comment:
You should probably return a `&str` here and let the caller decide if they want to allocate.

Reply:
Oh that's a good point.  This is a place I'm lacking rust knowledge.  I'll make this change.

Reply:
I ended up just returning the same value as `name()`, `Option<std::borrow::Cow<'static, str>>`.